amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/resolution/stages/ExtensionDomainElementMerge.scala

```diff
@@ -167,47 +168,49 @@ class ExtensionDomainElementMerge(restrictions: MergingRestrictions,
     }
   }
 
-  private def mergeByKeyValue(target: DomainElement,
-                              field: Field,
-                              key: KeyField,
-                              master: AmfArray,
-                              extension: AmfArray): Unit = {
+  private def mergeArraysByKey(target: DomainElement,
+                               field: Field,
+                               key: KeyField,
+                               master: AmfArray,
+                               extension: AmfArray): Unit = {
 
-    val asSimpleProperty                  = isSimpleProperty(key)
-    var existing: Map[Any, DomainElement] = buildElementByKeyMap(key, master)
-    var nullKey: Option[DomainElement]    = findElementWithNullKey(key, master)
+    val asSimpleProperty                                 = isSimpleProperty(key)
+    var existingElements: Map[Any, DomainElement]        = buildElementByKeyMap(key, master)
+    var existingNullKeyElements: Map[Obj, DomainElement] = findElementsWithNullKey(key, master)
```

Comment:
Add comment saying that "if we have multiple elements with null key we merge by `meta obj`"